### ISSUE TITLE
Compilation fails after disk full condition is resolved

### ISSUE THREAD

**Issue by docelic** [crystal-lang/crystal#10134]
Recently I had a situation where disk got full while Crystal program compilation was going on.
The error was, expectedly:

```
LLVM ERROR: IO failure on output stream: No space left on device
LLVM ERROR: IO failure on output stream: No space left on device
LLVM ERROR: IO failure on output stream: No space left on device
LLVM ERROR: IO failure on output stream: No space left on device
LLVM ERROR: IO failure on output stream: No space left on device
LLVM ERROR: IO failure on output stream: No space left on device
LLVM ERROR: IO failure on output stream: No space left on device
LLVM ERROR: IO failure on output stream: No space left on device
cc: error: U-nibilium5858X-wA-.o: No such file or directory
cc: error: S-pec5858E-qualE--105ff32cf1d08854d584bd47258d7dc8.o: No such file or directory
cc: error: S-pec5858E-qualE--0bd60a1583657679b0e2f8960921d528.o: No such file or directory
cc: error: S-pec5858E-qualE--392231f409d85494628279e72da6d6e8.o: No such file or directory
Error: execution of command failed with code: 1: `cc "${@}" -o /home/user/.cache/crystal/crystal-run-spec.tmp  -rdynamic -L/opt/crystal-0.35.1-1/bin/../lib/crystal/lib -L/usr/local/lib -lunibilium -lpcre -lm -lgc -lpthread /opt/crystal-0.35.1-1/share/crystal/src/ext/libcrystal.a -levent -lrt -ldl`
```

However, after the disk space issue was resolved, compilations wouldn't continue working until the cache directory `~/.cache/crystal/` was deleted:

```
$ crystal  s
/usr/bin/ld: _main.o: in function `->':
/home/docelic/p/labs/unibilium-shim.cr/spec/unibilium-shim_spec.cr:51: undefined reference to `*Unibilium::Terminfo#run<Pointer(UInt8), Int32, Int32>:Slice(UInt8)'
collect2: error: ld returned 1 exit status
Error: execution of command failed with code: 1: `cc "${@}" -o /home/docelic/.cache/crystal/crystal-run-spec.tmp  -rdynamic -L/opt/crystal-0.35.1-1/bin/../lib/crystal/lib -L/usr/local/lib -lunibilium -lpcre -lm -lgc -lpthread /opt/crystal-0.35.1-1/share/crystal/src/ext/libcrystal.a -levent -lrt -ldl
```

`rm -rf ~/.cache/crystal`

```
crystal  s
....

Finished in 811 microseconds
4 examples, 0 failures, 0 errors, 0 pending
```

Can this be improved, or we are aware of and accept the current behavior?

Thanks.

**Comment by straight-shoota** [comment]
This is likely already fixed on master (#9558).

**Comment by docelic** [comment]
Thanks, will reopen in case I don't manage to confirm.
